chia/wallet/conditions.py

```diff
@@ -0,0 +1,1221 @@
+            final_condition_list.append(AssertHeightRelative(self.min_blocks_since_created))
+        if self.min_height is not None:
+            final_condition_list.append(AssertHeightAbsolute(self.min_height))
+        if self.max_secs_after_created is not None:
+            final_condition_list.append(AssertBeforeSecondsRelative(self.max_secs_after_created))
+        if self.max_time is not None:
+            final_condition_list.append(AssertBeforeSecondsAbsolute(self.max_time))
+        if self.max_blocks_after_created is not None:
+            final_condition_list.append(AssertBeforeHeightRelative(self.max_blocks_after_created))
+        if self.max_height is not None:
+            final_condition_list.append(AssertBeforeHeightAbsolute(self.max_height))
+
+        return final_condition_list
+
+
+# Properties of the dataclass above, grouped by their property
+SECONDS_PROPERTIES: Set[str] = {"min_secs_since_created", "min_time", "max_secs_after_created", "max_time"}
+HEIGHT_PROPERTIES: Set[str] = {"min_blocks_since_created", "min_height", "max_blocks_after_created", "max_height"}
+AFTER_PROPERTIES: Set[str] = {"min_blocks_since_created", "min_height", "min_secs_since_created", "min_time"}
+BEFORE_PROPERTIES: Set[str] = {"max_blocks_after_created", "max_height", "max_secs_after_created", "max_time"}
+RELATIVE_PROPERTIES: Set[str] = {
+    "min_blocks_since_created",
+    "min_secs_since_created",
+    "max_secs_after_created",
+    "max_blocks_after_created",
+}
+ABSOLUTE_PROPERTIES: Set[str] = {"min_time", "max_time", "min_height", "max_height"}
+ALL_PROPERTIES: Set[str] = SECONDS_PROPERTIES | HEIGHT_PROPERTIES
+
+
+def parse_timelock_info(conditions: List[Condition]) -> ConditionValidTimes:
+    valid_times: ConditionValidTimes = ConditionValidTimes()
+    properties: Set[str] = ALL_PROPERTIES.copy()
+    for condition in conditions:
+        if isinstance(condition, TIMELOCK_DRIVERS):
+            timelock: Timelock = Timelock.from_program(condition.to_program())
+        elif isinstance(condition, Timelock):
+            timelock = condition
+        else:
+            # Something about python 3.9 makes this be not covered but on 3.10+ it is covered
+            # https://github.com/nedbat/coveragepy/issues/1530
+            continue  # pragma: no cover
+
+        properties_left = properties.copy()
+        min_not_max: bool = True
+        if timelock.after_not_before:
+            min_not_max = False
+            properties_left -= BEFORE_PROPERTIES
+        else:
+            properties_left -= AFTER_PROPERTIES
+
+        if timelock.seconds_not_height:
+            properties_left -= HEIGHT_PROPERTIES
+        else:
+            properties_left -= SECONDS_PROPERTIES
+
+        if timelock.relative_not_absolute:
+            properties_left -= ABSOLUTE_PROPERTIES
+        else:
+            properties_left -= RELATIVE_PROPERTIES
+
+        final_property: str = next(iter(properties_left))
```

Comment:
I think it might make sense to add:
```
assert(len(properties_left) == 1)
```
To ensure all the sets were defined correctly

Reply:
Added